src/services/services.ts

```diff
@@ -1363,6 +1363,11 @@ namespace ts {
                 onUnRecoverableConfigFileDiagnostic: noop,
             };
 
+            // The call to isProgramUptoDate below may refer back to documentRegistryBucketKey;

```

Comment:
That's scary, but I don't see a better way around it.  @sheetalkamat might.

Either way, thanks for the comment!

Reply:
Seems like its needed only for config file's json and we currently dont use that since we implement `getParsedCommandLine` on project..  But yeah this change looks good